enhancements/dns/coredns-egressfirewall-integration.md

```diff
@@ -0,0 +1,505 @@
+on the DNS name as it will not return the IP addresses of all the subdomains. The DNS lookup of the wildcard DNS name may fail to return
+any IP address as well. If the lookup for the wildcard DNS name fails, then it will retried using the default TTL (30 minutes). If the lookup
+succeeds then the details will be added to the `.status` of the corresponding CR. Thus, the lookups will be performed on the DNS names which
+are updated in the `.status` of the corresponding wildcard `EgressFirewallDNSName` CRs.
+
+The following `EgressFirewallDNSName` CRD will be added to the `dns.openshift.io` api-group.
+
+````go
+// EgressFirewallDNSName describes a DNS name used in a EgressFirewall rule. It is TechPreviewNoUpgrade only.
+type EgressFirewallDNSName struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// metadata is the standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec is the specification of the desired behavior of the EgressFirewallDNSName.
+	Spec EgressFirewallDNSNameSpec `json:"spec,omitempty"`
+	// status is the most recently observed status of the EgressFirewallDNSName.
+	Status EgressFirewallDNSNameStatus `json:"status,omitempty"`
+}
+
+// EgressFirewallDNSNameSpec is a desired state description of EgressFirewallDNSName.
+type EgressFirewallDNSNameSpec struct {
+	// name is the DNS name used in a EgressFirewall rule.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=^(\*\.)?([A-Za-z0-9-]+\.)*[A-Za-z0-9-]+\.$
+	Name string `json:"name"`
+}
+
+// EgressFirewallDNSNameStatus defines the observed status of EgressFirewallDNSName.
+type EgressFirewallDNSNameStatus struct {
+	// The list of matching DNS names and their corresponding IP addresses along with TTL and last
+	// DNS lookup time.
+	ResolvedNames []EgressFirewallDNSNameStatusItem `json:"resolvedNames,omitempty"`
+}
+
+// EgressFirewallDNSNameStatusItem describes the details of a resolved DNS name.
+type EgressFirewallDNSNameStatusItem struct {
+	// The resolved DNS name corresponding to the Name field of EgressFirewallDNSNameSpec.
+	// +kubebuilder:validation:Pattern=^(\*\.)?([A-Za-z0-9-]+\.)*[A-Za-z0-9-]+\.$
+	DNSName string `json:"dnsName"`
+	// The IP addresses associated with the DNS name used in a EgressFirewall rule.
+	IPs []string `json:"ips"`
+	// Minimum time-to-live value among all the IP addresses.
+	TTL int64 `json:"ttl"`
+	// Timestamp when the last DNS lookup was successfully completed.
+	LastLookupTime metav1.Time `json:"lastLookupTime"`
+}
+````
+
+### Workflow Description
+
+The workflows for Create, Delete and Update events for EgressFirewall related to DNS names are explained in this section. The workflow
+for those events is shown in the following diagram with an example `EgressFirewallDNSName` CR:
+
+![Workflow](./coredns-egressfirewall-integration.png)
```

Comment:
I can't figure out which field `Egress Firewall DNS Name Controller` actually updates?

Reply:
The Controller will remove stale DNS name entries from the status of the CRs. I have explained it here: https://github.com/openshift/enhancements/pull/1335/files#diff-b27f04c5e2673c47f922695835e746ecb2205a9f7800def32cc15c25af521758R276-R282